src/NuGet.Clients/NuGet.CommandLine/Commands/InstallCommand.cs

```diff
@@ -224,6 +224,20 @@ private async Task PerformV2RestoreAsync(string packagesConfigFilePath, string i
                     ClientPolicyContext = clientPolicyContext
                 };
 
+                IEnumerable<SourceRepository> sourceRepositories = packageRestoreContext.SourceRepositories;
+                IEnumerable<SourceRepository> enabledSources = (sourceRepositories != null && sourceRepositories.Any()) ? sourceRepositories :
+                    sourceRepositoryProvider.GetRepositories().Where(e => e.PackageSource.IsEnabled);
+
+                foreach (SourceRepository enabledSource in enabledSources)
+                {
+                    PackageSource source = enabledSource.PackageSource;
+                    if (source.IsHttp && !source.IsHttps)
+                    {
+                        projectContext.Log(RestoreLogMessage.CreateWarning(NuGetLogCode.NU1803,
+                            string.Format(CultureInfo.CurrentCulture, LocalizedResourceManager.GetString("Warning_HttpServerUsage"), "restore", source.Source)));
+                    }
+                }
+
                 var result = await PackageRestoreManager.RestoreMissingPackagesAsync(
```

Comment:
Is it possible to warn in PackageRestoreManager instead? That way you capture all scenarios.

Reply:
Call to `PackageRestoreManager` happen later for each packagereference, so it'll duplicate the warning.
That is why previously added `packageSourceMapping` logic just above here line 220ish too.

Reply:
Not sure I follow. In this codepath, it seems like the only place where this is called.